scripts/globals/additional_effects.lua

```diff
@@ -0,0 +1,263 @@
+
+    return damage
+end
+
+tpz.addEffect.levelCorrection = function(dLV, aLV, chance)
+    -- Level correction of proc chance (copied from existing bolt/arrow scripts, looks wrong..)
+    if dLV > aLV then
+        chance = utils.clamp(chance - 5 * (dLV - aLV), 5, 95)
+    end
+
+    return chance
+end
+
+tpz.addEffect.statusAttack = function(addStatus)
+    local effectList =
+    {
+        [tpz.effect.DEFENSE_DOWN] = {tick = 0, strip = tpz.effect.DEFENSE_BOOST},
+        [tpz.effect.EVASION_DOWN] = {tick = 0, strip = tpz.effect.EVASION_BOOST},
+        [tpz.effect.ATTACK_DOWN]  = {tick = 0, strip = tpz.effect.ATTACK_BOOST},
+        [tpz.effect.POISON]       = {tick = 3, strip = nil},
+        [tpz.effect.CHOKE]        = {tick = 3, strip = nil},
+    }
+    local delEffect = effectList[addStatus]
+    defender:delStatusEffect(delEffect.strip)
+    return tick
+end
+
+tpz.addEffect.calcDamage = function(attacker, element, defender, damage)
+    local params = {}
+    params.bonusmab = 0
+    params.includemab = false
+    damage = addBonusesAbility(attacker, element, defender, damage, params)
+    damage = damage * applyResistanceAddEffect(attacker, defender, element, 0)
+    damage = adjustForTarget(defender, damage, element)
+    damage = finalMagicNonSpellAdjustments(attacker, defender, element, damage)
+    return damage
+end
+
+-- paralyze on hit, fire damage on hit, etc..
+function additionalEffectAttack(attacker, defender, baseAttackDamage, item)
+    local addType = item:getMod(tpz.mod.ITEM_ADDEFFECT_TYPE)
+    local subEffect = item:getMod(tpz.mod.ITEM_SUBEFFECT)
+    local damage = item:getMod(tpz.mod.ITEM_ADDEFFECT_DMG)
+    local chance = item:getMod(tpz.mod.ITEM_ADDEFFECT_CHANCE)
+    local element = item:getMod(tpz.mod.ITEM_ADDEFFECT_ELEMENT)
+    local addStatus = item:getMod(tpz.mod.ITEM_ADDEFFECT_STATUS)
+    local power = item:getMod(tpz.mod.ITEM_ADDEFFECT_POWER)
+    local duration = item:getMod(tpz.mod.ITEM_ADDEFFECT_DURATION)
+    local msgID = 0
+    local msgValue = 0
+    local procType =
+    {
+        -- These are arbitrary, make up new ones as needed.
+        NORMAL = 1,
+        HP_HEAL = 2,
+        MP_HEAL = 3,
+        HP_DRAIN = 4,
+        MP_DRAIN = 5,
+        TP_DRAIN = 6,
+        HPMPTP_DRAIN = 7,
+        DISPEL = 8,
+        SELF_BUFF = 9,
+        DEATH = 10,
+    }
+
+    -- If we're not going to proc, lets not execute all those checks!
+    if chance < math.random(100) then
+        return 0,0,0
+    end
+
+    --------------------------------------
+    -- Modifications for proc's sourced from ranged attacks. See notes at top of script.
+    if tpz.addEffect.isRanged(item) then
+        if element then
+            damage = tpz.addEffect.calcRangeBonus(attacker, defender, element, damage)
+        end
+        chance = ltpz.addEffect.evelCorrection(defender:getMainLvl(), attacker:getMainLvl(), chance)
+    end
+    --------------------------------------
+
+    if addType == procType.NORMAL then
```

Comment:
In general on this if / elseif block, we could use tabled functions to check for which code to execute in a single operation. Instead of defining local vars from mods at the top, throw them into a table:
```lua
effectInfo = {
    addType = item:getMod(tpz.mod.ITEM_ADDEFFECT_TYPE),
    subEffect = item:getMod(tpz.mod.ITEM_SUBEFFECT),
    damage = item:getMod(tpz.mod.ITEM_ADDEFFECT_DMG),
}
```

Then you can toss them into functions tabled by addType:
```lua
effectActions = {
    procType.NORMAL = function(attacker, defender, effect)
        local msgID, msgValue
        if effect.addStatus and effect.addStatus > 0 then
            local tick = tpz.addEffect.statusAttack(effect.addStatus)
    --etc
        return subEffect, msgID, msgValue
    end
}
```

And then for `additionalEffectAttack` attack itself:
```lua
-- effect is table from above based on item mods
local effectAction = effectActions[item.addType]
subEffect, msgID, msgValue = effectAction(attacker, defender, effect)
return subEffect, msgID, msgValue
```

Reply:
you can see where I started to write a tabled for this, but I want to do further refactoring in the branch after the sql is in, but before hitting release. it got too big to handle alone in a single pr. Gonna take several.